src/mame/toaplan/dt7.cpp

```diff
@@ -141,10 +154,41 @@ uint8_t dt7_state::read_port_t()
 uint8_t dt7_state::read_port_2()
 {
 	logerror("%s: read port 2\n", machine().describe_context());
-
 	return 0xff;
```

Comment:
At least a few of the bits written to this port need to be read back here, since the code is using bit manipulation instructions and the V25 device doesn't enforce port directions.

Reply:
ahh, indeed, I think I had that in the code at one point, but this has been sat around for so long I must have forgot about it, anyway, it's added now.